metabase/metabase
# Utilize diagnoseExpression to detect expression errors

closes https://github.com/metabase/metabase/issues/38944
closes https://github.com/metabase/metabase/issues/38954
closes https://github.com/metabase/metabase/issues/39438

### Description

Invalid expressions are dropped by MLv2, before migration invalid expressions weren't dropped, but they couldn't be executed anyway, so this PR uses `diagnoseExpression` method from MLv2 to detect invalid expressions.

Currently, it's not possible to highlight the exact place with an error and get user-friendly error message from MLv2, so we use a generic message `Invalid expression`.

### How to verify

follow steps from the changed e2e test

or 

- Create a new question using the Sample Database > Orders table
- Add a custom column `case(isempty([Discount]), "true", "false")` called "discount is empty"
- Done button should be disabled and you should see a message "invalid expression"

### Demo

<img width="606" alt="image" src="https://github.com/metabase/metabase/assets/125459446/245b5192-e324-4d54-8f40-36e253885a99">


### Checklist
